internal/cli/uninstall.go

```diff
@@ -0,0 +1,204 @@
+			return 1
+		}
+		if err = clisnapshot.WriteSnapshot(ctx, c.project.Client(), w); err != nil {
+			fmt.Fprintf(os.Stderr, "Error generating snapshot: %s", err)
+			return 1
+		}
+		s.Update("Snapshot %q generated", c.snapshotName)
+	} else {
+		s.Update("skip-snapshot set; not generating server snapshot")
+		s.Status(terminal.StatusWarn)
+	}
+	s.Done()
+
+	// Uninstall
+	p, ok := serverinstall.Platforms[strings.ToLower(c.platform)]
+	if !ok {
+		c.ui.Output(
+			"Error uninstalling server from %s: invalid platform",
+			c.platform,
+			terminal.WithErrorStyle(),
+		)
+
+		return 1
+	}
+
+	err = p.Uninstall(ctx, &serverinstall.InstallOpts{
+		Log: log,
+		UI:  c.ui,
+	})
+	if err != nil {
+		c.ui.Output(
+			"Error uninstalling server from %s: %s\nSee Troubleshooting docs "+
+				"for guidance on manual uninstall: https://www.waypointproject.io/docs/troubleshooting",
+			c.platform,
+			clierrors.Humanize(err),
+			terminal.WithErrorStyle(),
+		)
+
+		return 1
+	}
+
+	// Post-uninstall cleanup of context
+	if c.deleteContext {
+		if err := c.contextStorage.Delete(contextDefault); err != nil {
+			c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+			return 1
+		}
+	}
+
+	c.ui.Output("Waypoint server successfully uninstalled for %s platform", c.platform, terminal.WithSuccessStyle())
+
+	return 0
+}
+
+func (c *UninstallCommand) AutocompleteArgs() complete.Predictor {
+	return complete.PredictNothing
+}
+
+func (c *UninstallCommand) AutocompleteFlags() complete.Flags {
+	return c.Flags().Completions()
+}
+
+func (c *UninstallCommand) Synopsis() string {
+	return "Uninstall the Waypoint server"
+}
+
+func (c *UninstallCommand) Help() string {
+	return formatHelp(`
+Usage: waypoint server uninstall [options]
+
+  Uninstall the Waypoint server. The platform should be specified as kubernetes,
+  nomad, or docker. '-auto-approve' is required.
+
+  By default, this command deletes the default server's context.
+
+  This command does not destroy Waypoint resources, such as deployments and 
+  releases. Clear all workspaces prior to uninstall to prevent hanging resources.
+
+` + c.Flags().Help())
+}
+
+func (c *UninstallCommand) Flags() *flag.Sets {
+	return c.flagSet(0, func(set *flag.Sets) {
+		f := set.NewSet("Command Options")
+		f.BoolVar(&flag.BoolVar{
+			Name:    "auto-approve",
+			Target:  &c.autoApprove,
+			Default: false,
+			Usage:   "Auto-approve server uninstallation.",
+		})
+
+		f.BoolVar(&flag.BoolVar{
+			Name:    "delete-context",
+			Target:  &c.deleteContext,
+			Default: true,
+			Usage:   "Delete the context for the server once it's uninstalled.",
+		})
+
+		f.StringVar(&flag.StringVar{
+			Name:    "platform",
+			Target:  &c.platform,
+			Default: "",
+			Usage:   "Platform to uninstall the Waypoint server from.",
+		})
+
+		f.StringVar(&flag.StringVar{
+			Name:    "snapshot-name",
+			Target:  &c.snapshotName,
+			Default: fmt.Sprintf("waypoint-sever-snapshot-%d", time.Now().Unix()),
```

Comment:
Ahhh, this is what I was talking about on my PR. I think this might be misleading in the help text, since the time will be different when the snapshot is taken? Maybe we can show the base name `waypoint-server-snapshot` here and mention in `Usage` that a timestamp will be applied when the snapshot is taken?

Reply:
Sure i'll update real quick to match!